### PR TITLE
fontforge 20200314

### PR DESCRIPTION
**Pull request by mistydemeo**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

fontforge has switched to cmake, so much of the formula had to get retooled. I can't get it to build against the system libxml2 anymore; it fails with this error, which references a path that comes from the `.pc` file we ship:

```
CMake Error in fontforge/CMakeLists.txt:
  Imported target "LibXml2::LibXml2" includes non-existent path

    "/usr/include/libxml2"

  in its INTERFACE_INCLUDE_DIRECTORIES.  Possible reasons include:

  * The path was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and references files it does not
  provide.
```

The Python audit failure is new, but I'm not familiar enough with the tooling to know how to fix this.

```
  * python modules have explicit framework links
    These python extension modules were linked directly to a Python
    framework binary. They should be linked with -undefined dynamic_lookup
    instead of -lpython or -framework Python.
      /usr/local/opt/fontforge/lib/python3.7/site-packages/fontforge.so
            /usr/local/opt/fontforge/lib/python3.7/site-packages/psMat.so
```

Closes #52292.

### CODE REVIEW

**Comment by EricFromCanada** [comment]
More info on `libxml2` being moved: Homebrew/brew#7278

**Comment by Bo98** [comment]
Ah yeah, *.pc files have also been updated so the libxml2 part of this should now work.

**Comment by mistydemeo** [comment]
Perfect - reverted my libxml2 changes and rebased, and it's building against the system libxml2 now. I'll try to get the Python extensions sorted this weekend.

**Comment by mistydemeo** [comment]
Looks like the failure is because both Python 3.7 (via a direct dep) and 3.8 (via recursive dependencies) are in the tree; cmake is picking 3.8.

File: Formula/fontforge.rb [hunk 1]
```diff
@@ -67,8 +57,8 @@ def caveats
   test do
     system bin/"fontforge", "-version"
     system bin/"fontforge", "-lang=py", "-c", "import fontforge; fontforge.font()"
-    xy = Language::Python.major_minor_version "python3"
+    xy = Language::Python.major_minor_version "python3@3.8"
```
Comment (bayandin) [inline Formula/fontforge.rb:60]:
```suggestion
    xy = Language::Python.major_minor_version Formula["python@3.8"].opt_bin/"python3"
```

**Comment by iMichka** [comment]
Superseded by #54443

**Comment by Lesha17** [comment]
Hello!

I've updated Homebrew, and I have the same error while installing llvm following this instruction https://github.com/rust-lang/rust/issues/63417#issuecomment-520181587 after installing libxml2. This issue happens with python3.7 and python3.9

**Pull request closed without merging**